src/mono/mono/metadata/custom-attrs.c

```diff
@@ -2056,6 +2057,53 @@ mono_custom_attrs_from_method_checked (MonoMethod *method, MonoError *error)
 	return mono_custom_attrs_from_index_checked (m_class_get_image (method->klass), idx, FALSE, error);
 }
 
+gboolean
+mono_method_get_unsafe_accessor_attr_data (MonoMethod *method, int *accessor_kind, char **member_name, MonoError *error)
+{
+	MonoCustomAttrInfo *cinfo = mono_custom_attrs_from_method_checked (method, error);
+
+	if (!cinfo)
+		return FALSE;
+
+	MonoClass *unsafeAccessor = mono_class_try_get_unsafe_accessor_attribute_class ();
+	MonoCustomAttrEntry *attr = NULL;
+
+	for (int idx = 0; idx < cinfo->num_attrs; ++idx) {
+		MonoClass *ctor_class = cinfo->attrs [idx].ctor->klass;
+		if (ctor_class == unsafeAccessor) {
+			attr = &cinfo->attrs [idx];
+			break;
+		}
+	}
+
+	if (!attr)
+		return FALSE;
+
+	MonoDecodeCustomAttr *decoded_args = mono_reflection_create_custom_attr_data_args_noalloc (m_class_get_image (attr->ctor->klass), attr->ctor, attr->data, attr->data_size, error);
+	
+	if (!is_ok (error)) {
+		mono_error_cleanup (error);
```

Comment:
Ditto.

Reply:
I added code to free `decoded_args`. Or did you mean another variable?

Reply:
Oh I need to free it before return.